app/src/main/java/com/samco/trackandgraph/ui/CustomiseCardDialog.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ *  This file is part of Track & Graph
+ *
+ *  Track & Graph is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Track & Graph is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Track & Graph.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.samco.trackandgraph.ui
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import com.samco.trackandgraph.database.*
+import com.samco.trackandgraph.database.entity.*
+import com.samco.trackandgraph.databinding.CustomiseCardDialogBinding
+import kotlinx.coroutines.*
+
+const val CUSTOMISE_CARD_DIALOG_FEATURE_KEY = "customise_card_group"
+
+class CustomiseCardDialogFragment : DialogFragment() {
+    private val viewModel by viewModels<CustomiseCardDialogViewModel>()
+    private lateinit var binding: CustomiseCardDialogBinding
+    private val featurePeriodList = listOf(
+        FeatureShowCountPeriod.ALL,
+        FeatureShowCountPeriod.DAILY,
+        FeatureShowCountPeriod.WEEKLY,
+        FeatureShowCountPeriod.MONTHLY,
+        FeatureShowCountPeriod.YEARLY
+    )
+    private val featureMethodList = listOf(
+        FeatureShowCountMethod.COUNT_ENTRIES,
+        FeatureShowCountMethod.SUM_VALUES
+    )
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = CustomiseCardDialogBinding.inflate(inflater, container, false)
+        return activity?.let {
+            val id = requireArguments().getLong(CUSTOMISE_CARD_DIALOG_FEATURE_KEY)
+            viewModel.init(requireActivity(), id)
```

Comment:
I don't think you need this extra scope, requireActivity will throw an exception for you